adb_graphics/datahandler/gribdata.py

```diff
@@ -481,128 +281,185 @@ class fieldData(UPPData):
 
-        flru = np.where((ceil > 1.) & (ceil < 3.), 1.01, 0.0)
-        flru = np.where((vis > 3.) & (vis < 5.), 1.01, flru)
-        flru = np.where((ceil > 0.5) & (ceil < 1.), 2.01, flru)
-        flru = np.where((vis > 1.) & (vis < 3.), 2.01, flru)
-        flru = np.where((ceil > 0.0) & (ceil < 0.5), 3.01, flru)
-        flru = np.where((vis < 1.), 3.01, flru)
+        flru = where((ceil > 1.0) & (ceil < 3.0), 1.01, 0.0)
+        flru = where((vis > 3.0) & (vis < 5.0), 1.01, flru)
+        flru = where((ceil > 0.5) & (ceil < 1.0), 2.01, flru)
+        flru = where((vis > 1.0) & (vis < 3.0), 2.01, flru)
+        flru = where((ceil > 0.0) & (ceil < 0.5), 3.01, flru)
+        flru = where((vis < 1.0), 3.01, flru)
 
         vis.close()
 
-        return xr.DataArray(flru)
+        return DataArray(flru)
 
     @property
     def cmap(self):
+        """
+        The LinearSegmentedColormap specified by the config key 'cmap'.
+        """
 
-        ''' Returns the LinearSegmentedColormap specified by the config key
-        "cmap" '''
-
-        return cm.get_cmap(self.vspec['cmap'])
+        return get_cmap(self.vspec["cmap"])
 
     @property
     def colors(self) -> np.ndarray:
-
-        '''
-        Returns a list of colors, specified by the config key "colors".
-
-        The yaml file "colors" key may contain a list or a function to be
-        called.
-        '''
-
-        color_spec = self.vspec.get('colors')
-
-        if isinstance(color_spec, (list, np.ndarray)):
-            return np.asarray(color_spec)
+        """
+        Returns an array of colors, specified by the config key "colors".
+        """
+
+        color_spec = self.vspec.get("colors", "")
+        if not color_spec:
+            msg = f"No colors definition found for {self.short_name} at {self.level}"
+            raise errors.NoGraphicsDefinitionForVariableError(msg)
         try:
             ret = self.__getattribute__(color_spec)
-            if callable(ret):
-                return ret()
-            return ret
-        except AttributeError:
-            return color_spec
+        except AttributeError as e:
+            msg = f"There is no color definition named {color_spec}"
+            raise AttributeError(msg) from e
+        if callable(ret):
+            return np.asarray(ret())
+        return np.asarray(ret)
 
     @property
     def corners(self) -> list:
+        """
 
-        '''
-        Returns lat and lon of lower left (ll) and upper right(ur) corners:
+        Returns lat and lon of lower left (ll) and upper right(ur) corners.
+
+        Order:
                ll_lat, ur_lat, ll_lon, ur_lon
-        '''
+        """
 
         lat, lon = self.latlons()
-        if self.model in ['global', 'hfip', 'obs']:
-            ret = [lat[-1], lat[0], lon[0], lon[-1]]
-        elif self.model == 'global_mpas':
-            ret = [lat[0], lat[-1], lon[0], lon[-1]]
-        else:
-            ret = [lat[0, 0], lat[-1, -1], lon[0, 0], lon[-1, -1]]
```

Comment:
This was the logic that was replaced by the flipping of the lats in the `latlons` method.